src/projects.rs

```diff
@@ -67,35 +67,37 @@ fn fetch_next_item(config: Config, project_name: &str) -> Result<Option<Item>, S
     Ok(items.first().map(|item| item.to_owned()))
 }
 
-/// Get next items and give an interactive prompt for completing them
-pub fn next_item_interactive(config: Config, project_name: &str) -> Result<String, String> {
-    let mut config = config;
-    loop {
-        match fetch_next_item(config.clone(), project_name) {
-            Ok(Some(item)) => {
-                config.set_next_id(item.id.clone()).save()?;
-                config = Config::load(&config.path)?;
-                match handle_item(config.clone(), item) {
-                    Some(Ok(_)) => (),
-                    Some(Err(e)) => return Err(e),
-                    None => return Ok(green_string("Exited")),
-                }
-            }
-            Ok(None) => return Ok(green_string("Done")),
-            Err(e) => return Err(e),
+/// Get next items and give an interactive prompt for completing them one by one
+pub fn process_items(config: Config, project_name: &str) -> Result<String, String> {
+    let project_id = projects::project_id(&config, project_name)?;
+    let items = request::items_for_project(&config, &project_id)?;
+    let items = items::filter_not_in_future(items, &config)?;
+    for item in items {
+        config.set_next_id(item.id.clone()).save()?;
+        match handle_item(config.reload()?, item) {
+            Some(Ok(_)) => (),
+            Some(Err(e)) => return Err(e),
+            None => return Ok(green_string("Exited")),
         }
     }
+    Ok(green_string(&format!(
+        "There are no more tasks in '{}'",
+        project_name
+    )))
 }
```

Comment:
You can now use the new string interpolation syntax


```suggestion
    Ok(green_string(&format!(
        "There are no more tasks in '{project_name}'"
    )))
}
```